openapi3filter/validate_response.go

```diff
@@ -159,11 +160,29 @@ func ValidateResponse(ctx context.Context, input *ResponseValidationInput) error
 
 	// Validate data with the schema.
 	if err := contentType.Schema.Value.VisitJSON(value, append(opts, openapi3.VisitAsResponse())...); err != nil {
+		schemaId := getSchemaIdentifier(contentType.Schema)
 		return &ResponseError{
 			Input:  input,
-			Reason: "response body doesn't match the schema",
+			Reason: fmt.Sprintf("response body doesn't match the schema %s", schemaId),
 			Err:    err,
+			// TODO: add an error message customizer function similar to SchemaError
 		}
 	}
 	return nil
 }
+
+func getSchemaIdentifier(schema *openapi3.SchemaRef) string {
+	var id string
+
+	if schema != nil {
+		id = schema.Ref
+	}
+	if strings.TrimSpace(id) == "" && schema.Value != nil {
+		id = schema.Value.Title
+	}
+	if strings.TrimSpace(id) == "" && schema.Value != nil {
+		id = schema.Value.Description
```

Comment:
> description - [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.

This field can be many lines long. Aren't you overfitting to your problem space here? I don't think this is a good idea.

Reply:
I had that some concern but erred on the side of caution. Now that I see you share that concern I agree description should be removed. This is fixed in upcoming commit